src/main/java/org/kohsuke/github/GitHub.java

```diff
@@ -854,7 +854,169 @@ public GHAuthorization createToken(Collection<String> scope, String note, String
+        server.setExecutor(null); // creates a default executor
+        server.start();
+        
+        
+		String doRequest = "https://github.com/login/oauth/authorize?" +
+		"client_id=" + OAuthApplicationclient_id + "&"	
+		+ "redirect_uri=http%3A%2F%2Flocalhost%3A"+WEBSERVER_PORT+"%2Fsuccess" + "&" +
+		"response_type=code" + "&" + 
+		(login!=null?"login="+login.replaceAll("@", "%40") + "&":"") + 
+		"allow_signup=true" + "&" + 
+		"state="+state + "&" +
+		"scope=";
+        int i=0;
+		for (String s:scope) {
+			s = s.replaceAll(":", "%3A");
+			doRequest += s ;
+			if(i!=scope.size()-1)
+				doRequest +=  "%20";
+			i++;
+		}
+		doRequest = doRequest.trim();
+		// Send request in step 1
+		// https://developer.github.com/apps/building-oauth-apps/authorizing-oauth-apps/#1-request-a-users-github-identity
+		// User interaction is needed to approve the authorization
+		// Open this URL in a desktop browser
+		if (java.awt.Desktop.isDesktopSupported() && java.awt.Desktop.getDesktop().isSupported(java.awt.Desktop.Action.BROWSE))
+			java.awt.Desktop.getDesktop().browse(new java.net.URI(doRequest));
+		
+        
+        // block here until webflow complete
+        long start = System.currentTimeMillis();
+		// 200 second timeout
+		while (System.currentTimeMillis() - start < 200 * 1000 && handler.tempCode == null) {
+			try {
+				Thread.sleep(100);
+			} catch (InterruptedException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+				break;
+			}
+
+		}
+		server.stop(0);
+        return createOAuthTokenWebFlowStepTwo(OAuthApplicationclient_id,OAuthApplicationSecret,handler.tempCode,state,redirect_after_auth);
+    }
+    
+    /**
+     * Creates a new authorization.
+     * <p>
+     * The token created can be then used for {@link GitHub#connectUsingOAuth(String)} in the future.
+     * 
+	 * @param state the unguessable random string you provided in Step 1. See https://developer.github.com/apps/building-oauth-apps/authorizing-oauth-apps/#2-users-are-redirected-back-to-your-site-by-github
+	 * 
+     * @param OAuthApplicationID the public ID for the OAuth application that is requesting the token
+     * 
+     * @param OAuthApplicationSecret the secret code passed to GitHub to identify the application
+     * @return the gh authorization
+     * @throws IOException
+     *             the io exception
+     * @see <a href="https://developer.github.com/apps/building-oauth-apps/authorizing-oauth-apps/#2-users-are-redirected-back-to-your-site-by-github">Documentation</a>
+     * 
+     */
+    public GHAuthorization createOAuthTokenWebFlowStepTwo( 
+    		@Nonnull  String OAuthApplicationclient_id,@Nonnull  String OAuthApplicationSecret,
+    		@Nonnull String temp_OAuth_Code,
+    		String state,
+    		java.net.URL redirect_after_auth) 
+    				throws IOException {
+        Requester requester = createRequest()
+        						.setRawUrlPath("https://github.com/login/oauth/access_token")
```

Comment:
@madhephaestus 

At minimum you're missing this:
```suggestion
        						.method("POST")
        						.setRawUrlPath("https://github.com/login/oauth/access_token")
```